fastplotlib/graphics/_base.py

```diff
@@ -79,3 +80,222 @@ def __repr__(self):
             return f"'{self.name}' fastplotlib.{self.__class__.__name__} @ {hex(id(self))}"
         else:
             return f"fastplotlib.{self.__class__.__name__} @ {hex(id(self))}"
+
+
+class Interaction(ABC):
+    @abstractmethod
+    def _set_feature(self, feature: str, new_data: Any, indices: Any):
+        pass
+
+    @abstractmethod
+    def _reset_feature(self, feature: str):
+        pass
+
+    def link(self, event_type: str, target: Any, feature: str, new_data: Any, callback_function: callable = None):
+        if event_type in self.pygfx_events:
+            self.world_object.add_event_handler(self.event_handler, event_type)
+        elif event_type in self.feature_events:
+            feature = getattr(self, event_type)
```

Comment:
aren't the even types for features named like "colors-changed" and not "color". Maybe you can just change the event type to be like "color" instead of "color-changed"? For example see: 

https://github.com/kushalkolar/fastplotlib/blob/master/fastplotlib/graphics/features/_colors.py#L192

Reply:
wait, can you clarify? you want the event_type for feature events to be "colors" or "colors-changed"?

Reply:
for now I changed it to "colors", "data", etc. in order to continue getattr() but can change if you want

Reply:
sounds good, simpler and minimize the parsing is better.